pgmpy/estimators/ExpertKnowledge.py

```diff
@@ -0,0 +1,41 @@
+class ExpertKnowledge:
+    """
+    Class to specify expert knowledge for causal discovery algorithms.
+    Currently, expert knowledge can provide information about edges that have to be present/absent in
+    the final learned graph and a limited search space for edges.
+    Parameters
+    ----------
+    white_list: list or None
+            If a list of edges is provided as `white_list`, the search is limited to those
+            edges. The resulting model will then only contain edges that are in `white_list`.
+            Default: None
+    black_list: list or None
+            If a list of edges is provided as `black_list`, they are excluded from the search
+            and the resulting model will not contain any of those edges. Default: None
+    fixed_edges: iterable
+            A list of edges that will always be there in the final learned model.
+            The algorithm will add these edges at the start of the algorithm and
+            will never change it.
+    """
+
+    def _validate_edges(self, edge_list):
+        if not hasattr(edge_list, "__iter__"):

```

Comment:
I would also suggest checking if an iterable object other than `set` is passed here. In that case, we can covert that to a set.

Reply:
I think that case would automatically be considered valid, since we are only rejecting non-iterables? A relevant example I saw was [this part of Mmhc estimator](https://github.com/pgmpy/pgmpy/blob/cf8d0f12e2e5be62b01ff8fded85f3f64eab1e84/pgmpy/estimators/MmhcEstimator.py#L88). Here the white list passed is of type <class 'networkx.classes.reportviews.OutEdgeView'>. So, any iterator is being accepted for each edge information right now. What do you think?

Reply:
Indeed it would work, but we do convert everything to a `set` in HillClimbSearch: https://github.com/pgmpy/pgmpy/blob/cf8d0f12e2e5be62b01ff8fded85f3f64eab1e84/pgmpy/estimators/HillClimbSearch.py#L299 . Doing that conversion here would just separate out this preprocessing logic.